fontforgeexe/cvstroke.c

```diff
@@ -170,36 +170,18 @@ static int _Stroke_OK(StrokeDlg *sd,int isapply) {
 	    err = true;
 	} else {
 	    SplineSet *ss;
-	    SplinePoint *sp;
 	    SplinePointListSelect(si->nib, false);
-	    int cnt, selectall = false;
-	    msg = NULL;
 	    for ( ss=si->nib ; ss!=NULL && !err; ss=ss->next ) {
-		if ( !err ) {
-		    enum ShapeType pt = NibIsValid(ss);
-		    if ( pt!=Shape_Convex ) {
-			msg = NibShapeTypeMsg(pt);
-			err = true;
-		    }
-		}
-		if ( selectall ) {
-		    for ( sp=ss->first;;) {
-			sp->selected = true;
-			if ( sp->next==NULL )
-		    break;
-			sp = sp->next->to;
-			if ( sp==ss->first )
-		    break;
-		    }
-		}
-		if ( err ) {
+		enum ShapeType pt = NibIsValid(ss);
+		if ( pt!=Shape_Convex ) {
```

Comment:
So this branch always represents an error condition?

Reply:
If the contour isn't `Shape_Convex`, yes, that's an error. 

This code had a bunch of extra checks and sub-parts left over from when it was doing more things. The `!err` in the `for` guard is all that's needed at this point to stop further checks. The `if (!err)` added below is just to make sure we only call `GDrawRequestExpose()` once (because the nib selected points may change, so it's good to repaint them).